src/posix/platform/radio_spinel.cpp

```diff
@@ -1073,6 +1112,18 @@ void RadioSpinel::RadioTransmit(void)
                     mTransmitFrame->mInfo.mTxInfo.mMaxCsmaBackoffs, mTransmitFrame->mInfo.mTxInfo.mMaxFrameRetries,
                     mTransmitFrame->mInfo.mTxInfo.mCsmaCaEnabled);
 
+    if (mTransmitFrame->mInfo.mTxInfo.mCsmaCaEnabled)
+    {
+        mTxRadioEndUs = otSysGetTime() + MIN_TX_WAIT_US +
+                        mTransmitFrame->mInfo.mTxInfo.mMaxFrameRetries *
+                            (1UL << (MAC_MIN_BE + mTransmitFrame->mInfo.mTxInfo.mMaxFrameRetries +
+                                     mTransmitFrame->mInfo.mTxInfo.mMaxCsmaBackoffs));
+    }
+    else
+    {
+        mTxRadioEndUs = otSysGetTime() + MIN_TX_WAIT_US;
```

Comment:
How about setting `mTxRadioEndUs` only when `error` is OT_ERROR_NONE?

Reply:
Updated accordingly